crates/pallet-executor/src/lib.rs

```diff
@@ -608,6 +608,12 @@ impl<T: Config> Pallet<T> {
 
         let (_, mut best_number) = <ReceiptHead<T>>::get();
 
+        if let Some(first_primary_number) = execution_receipts.get(0).map(|r| r.primary_number) {
+            if first_primary_number < best_number {
+                return Err(ExecutionReceiptError::Pruned);
```

Comment:
Commit message talks about increasing votes, but the code just rejects the bundle. I'm confused.

Reply:
Let's say the current receipt head is block 5, the oldest receipt is 3 (receipt {0,1, 2} have been pruned), the primary block is 7, for any reason, 

- if a bundle with receipts {4, 5, 6} was submitted, then the vote for receipt {4,5} will be increased, a new receipt 6 is appended.
- if a bundle with receipts {2, 3, 4, 5, 6} was submitted, then this bundle will be rejected as the first one has been pruned.

That said, given a bundle with multiple receipts ranging from an older to a recent block, if the receipt is not pruned and is already seen on the primary chain, the receipt votes will be increased accordingly, if the receipt is new, it will extend the receipt chain.

The part of increasing the vote(`apply_execution_receipt`) is already there, this change is primarily to reject the bundle including a pruned receipt. The increasing votes part in the commit message was intended to provide some more context but seemingly cause some more confusion instead :).